server/controllers/notifications.js

```diff
@@ -0,0 +1,47 @@
+const asyncHandler = require('express-async-handler');
+const Notifications = require('../models/Notifications');
+
+exports.getAllnotifications = asyncHandler(async(req, res) => {
+    const userId = req.params.id;
+    const notifications = await Notifications.find({
+        receiver: userId,
+
+    })
+    if (!notifications) {
+        res.status(404);
+        throw new Error('No notifications')
+    }
+    res.status(200).json({ notifications });
+
+});
+
+exports.readNotification = asyncHandler(async(req, res) => {
+    const notification = await Notifications.findByIdAndUpdate(req.params.id, { read: true }, { new: true });
+    if (!notification) {
+        res.status(404);
+        throw new Error('Cannot find notification');
+    }
+    res.status(200).json({ notification });
+
+});
+
+exports.createNotification = (async(req, res) => {
+    const notification = req.body;
+    const doc = await Notifications.insertMany({...notification, createdAt: Date.now() });
```

Comment:
insertMany is used when we have an array of documents to be stored into a collection. In our case, we are only creating one notification at a time, so you can modify the controllers accordingly :)

For additional reference, please see links: 
Creating and saving a single documents: https://www.tutorialkart.com/nodejs/mongoose/insert-document-to-mongodb/

MongoDB insertMany: https://docs.mongodb.com/manual/reference/method/db.collection.insertMany/